src/emulator/modules/SceCtrl/SceCtrl.cpp

```diff
@@ -227,7 +259,11 @@ static int peek_buffer_positive(HostState &host, int port, SceCtrlData *&pad_dat
         pad_data->ry = float_to_byte(axes[3]);
     }
 
-    return 0;
+    if (negative) {
+        pad_data->buttons = 0xFFFFFFFF - pad_data->buttons;
```

Comment:
no need to fix, just fyi you could do this here (less error prone, works for all sizes)
```suggestion
        pad_data->buttons = ~pad_data->buttons;
```